Update the ml/pre-processing

hi-T0day opened the pull request:
1. I rename the variable name. Now, the variable name is in the same
format and easy to understand.
2. I add the objdump processing code.
3. I combine all features and labels in one RDD, that is suitable for
tensor flow work.

hi-T0day committed:
Merge remote-tracking branch 'HolmesProcessing/master'

hi-T0day committed:
Update the ml/pre-processing

1. I rename the variable name. Now, the variable name is in the same
format and easy to understand .
2. I add the objdump processing code.
3. I combine all features and labels in one RDD, that is suitable for
tensor flow work.

webstergd commented on ml/pre-processing/get_features_from_peinfo.scala:
```diff
@@ -0,0 +1,56 @@
+}
+
+val List17 = Array.iterate(0.0,17)(a=>a*0)
+def findAllIntinPEINFO( peinfo_json_results : JsLookupResult, time: Double): Array[Double]= {
```
how do you handle multiple sections of the same type? Do you account for the other section types or just discard?

hi-T0day replied:
Hey, George. Sorry to reply you later.
I only extract four kinds of sections (.text / .data / .rsrc / .rdata). The other section are discarded. 
I do this because the code of Bojan's also only extracts these four kinds of sections and discards others.
What do you think about the sections with your experience?  It is not hard to add other sections into my code. 
BTW, I might have deflected your question because I dont understand this express in your words.
> multiple sections of the same type.

My understanding is .text section and .data section are multiple sections, however, I dont know the meaning of the same type.

webstergd commented on ml/pre-processing/get_VT_signatures.scala:
```diff
@@ -0,0 +1,45 @@
+    return regex.findFirstIn(x).isEmpty
+}
+
+val VT_results_by_service_name_meta = sc.cassandraTable[VT_results_by_service_name_class]("gsoc3","results_meta_by_service_name").where("service_name=?","virustotal")
```
they keyspace should be a configuration option

hi-T0day replied:
Yes, I plan to have a configuration file. But I failed.
The code is this:
```
import java.io.BufferedInputStream
import java.io.File
import java.io.FileInputStream
import java.io.InputStream
import org.apache.hadoop.conf._
import org.apache.hadoop.fs._
val conf = new Configuration()
val fileSystem = FileSystem.get(conf)
val stream = fileSystem.open(new Path("Configuration_File.txt"))
def readLines = Stream.cons(stream.readLine, Stream.continually( stream.readLine))
readLines.takeWhile(_ != null).foreach(line => println(line))
```
I plan to read configuration file by scala readfile method. However, there is this line in the code.
```
val conf = new Configuration()
```
Because I new the Configuration(), the spark interpreter configuration is covered and then the spark works with error. 
I try to solve this issue by getting spark configuration instead of newing configuration. But I cannot google the answer.

webstergd replied:
take a look at @cli0's pull #11

hi-T0day replied:
Thanks, I got it! I add the config file now. And I also change the code, replacing the keyspace, table, and file directories with parameters in config file.

webstergd commented on ml/pre-processing/get_VT_signatures.scala:
```diff
@@ -0,0 +1,45 @@
+val VT_join_results = VT_results_by_service_name_rdd.join(VT_results_by_sha256_rdd).map(x => (new VT_join_results_class(x._1._1,x._1._2, unzip(x._2._2.results)))).distinct().cache()
+val sample_signatures_rdd = VT_join_results.flatMap(x=>Json.parse(x.results) \ "scans" \\ "result").map(x=>Json.stringify(x)).filter( x=> !(x == "null"))
+val sample_signatures_split_rdd = sample_signatures_rdd.flatMap(x=>x.replaceAll("""["]""","").replaceAll("""\![a-zA-Z0-9\s\+]+""","").replaceAll("""@[a-zA-Z0-9\s\+]+""","").replaceAll("""~[a-zA-Z0-9\s\+]+""","").replaceAll("""[\(|\[|{][a-zA-Z0-9\s\+]*[\)|\]|}]""","").replaceAll("""(\.|\!|\:|\_|\-|\\|/|\[|\])"""," ").split(" ")).filter(x=>(x.size>3)).filter(x=>deleteNumberInSampleSignatures(x)).ma ...
+val signatures_prefix_rdd = sc.textFile("./t0day/prefix.txt").map(x=>x.toLowerCase())
```
what is the ./t0day/frefix.txt file? This needs to compile on its own

hi-T0day replied:
The prefix.txt file is the prefix and suffix of the VT results that Bojan sent me.
Considering some words in VT results are too normal (for example Malware, TrojWare), we should delete it. 
So this file does not need to compile and I just save this file. 
What do you think of making a new directory in ml/pre-processing to save prefix.txt file?

webstergd commented on ml/pre-processing/get_VT_signatures.scala:
```diff
@@ -0,0 +1,45 @@
+}
+
+val VT_sample_signatures_final_array_rdd = VT_sample_signatures_initial_seq_rdd.map(x=>new VT_sample_signatures_final_array_rdd_class(x.sha256,findAllInFamilySignatures(x.seq_results)))
+VT_sample_signatures_final_array_rdd.toDF().write.format("parquet").save("./t0day/VT_sample_signatures_final_array.parquet")
```
the save location should be a configuration option

hi-T0day committed:
Renew some bugs about peinfo

I forget to save the int features of peinfo.
So I concat the int features and binary features of peinfo and then
save them.

hi-T0day committed:
Add the config file

I add the PreProcessingConfig.scala file and replace file directory
with parameters in the config file.

webstergd commented:
Sweet thanks for the fixes dude. Just waiting on @bojankolosnjaji to give me the thumbs up or down
